rxnetty/src/main/java/io/reactivex/netty/client/ConnectionPoolImpl.java

```diff
@@ -110,75 +123,100 @@ public ConnectionPoolImpl(RxClient.ServerInfo serverInfo, PoolConfig poolConfig,
+                    return;
+                }
                 try {
-                    metricEventsSubject.onEvent(ClientMetricsEvent.POOL_ACQUIRE_START);
-                    PooledConnection<I, O> idleConnection = getAnIdleConnection(true);
-
-                    if (null != idleConnection) { // Found a usable connection
-                        idleConnection.beforeReuse();
-                        channelFactory.onNewConnection(idleConnection, subscriber);
-                        long endTime = Clock.onEndMillis(startTimeMillis);
-                        metricEventsSubject.onEvent(ClientMetricsEvent.POOLED_CONNECTION_REUSE, endTime);
-                        metricEventsSubject.onEvent(ClientMetricsEvent.POOL_ACQUIRE_SUCCESS, endTime);
-                    } else if (limitDeterminationStrategy.acquireCreationPermit(startTimeMillis,
-                                                                                TimeUnit.MILLISECONDS)) { // Check if it is allowed to create another connection.
-                        /**
-                         * Here we want to make sure that if the connection attempt failed, we should inform the strategy.
-                         * Failure to do so, will leak the permits from the strategy. So, any code in this block MUST
-                         * ALWAYS use this new subscriber instead of the original subscriber to send any callbacks.
-                         */
-                        Subscriber<? super ObservableConnection<I, O>> newConnectionSubscriber =
-                                newConnectionSubscriber(subscriber, startTimeMillis);
-                        try {
-                            channelFactory.connect(newConnectionSubscriber, serverInfo,
-                                                   connectionFactory); // Manages the callbacks to the subscriber
-                        } catch (Throwable throwable) {
-                            newConnectionSubscriber.onError(throwable);
-                        }
-                    } else { // Pool Exhausted
-                        PoolExhaustedException e = new PoolExhaustedException();
-                        metricEventsSubject.onEvent(ClientMetricsEvent.POOL_ACQUIRE_FAILED,
-                                                    Clock.onEndMillis(startTimeMillis), e);
-                        subscriber.onError(e);
+                    if (isShutdownRequested.get()) {
+                        subscriber.onError(new IllegalStateException("Connection pool is already shutdown."));
+                        return;
                     }
-                } catch (Throwable throwable) {
-                    metricEventsSubject.onEvent(ClientMetricsEvent.POOL_ACQUIRE_FAILED,
-                                                Clock.onEndMillis(startTimeMillis), throwable);
-                    subscriber.onError(throwable);
+                    performAquire(subscriber);
+                }   finally {
+                    lock.unlock();
```

Comment:
That is true but I increment the connections counter before I release the lock (in the previous commit it was a different counter but that is actually not needed). So if the lock is released the shutdown can not take place because the counter got incremented.